go/tendermint/tendermint.go

```diff
@@ -515,15 +519,28 @@ func (t *tendermintService) lazyInit() error {
 	tenderConfig.P2P.ListenAddress = viper.GetString(CfgCoreListenAddress)
 	tenderConfig.P2P.ExternalAddress = viper.GetString(cfgCoreExternalAddress)
 	tenderConfig.P2P.AllowDuplicateIP = true // HACK: e2e tests need this.
+	tenderConfig.P2P.PrivatePeerIDs = viper.GetString(CfgP2PPrivatePeers)
+	// Persistent peers need to be lowecase as p2p/transport.go:MultiplexTransport.upgrade()
+	// uses a case sensitive string comparision to validate public keys.
+	// Since persistent peers is expected to be in comma-delimited id@host:port format,
+	// lowercasing the whole string is ok.
+	tenderConfig.P2P.PersistentPeers = strings.ToLower(viper.GetString(CfgP2PPersistentPeers))
+	tenderConfig.P2P.PexReactor = !viper.GetBool(CfgP2PDisablePeerExchange)
 	tenderConfig.P2P.SeedMode = viper.GetBool(CfgP2PSeedMode)
 	// Seed Ids need to be Lowecase as p2p/transport.go:MultiplexTransport.upgrade()
-	// uses a case sensitive string comparision to validate public keys
+	// uses a case sensitive string comparision to validate public keys.
 	// Since Seeds is expected to be in comma-delimited id@host:port format,
 	// lowercasing the whole string is ok.
 	tenderConfig.P2P.Seeds = strings.ToLower(viper.GetString(CfgP2PSeeds))
```

Comment:
I there there were some plans in making `CfgP2PSeeds` a string slice (on our end) instead of comma-delimited list, maybe a good time to change it

Reply:
Done.

Reply:
I've also decided to make a breaking configuration change here and renamed `tendermint.seeds` configuraiton flag to `tendermint.seed` to be consistent with newly added `tendermint.*peer` options.

@ptrus, please take a look.